pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/ImmutableSegment.java

```diff
@@ -55,4 +55,13 @@ public interface ImmutableSegment extends IndexSegment {
    * @return Size of the segment in bytes
    */
   long getSegmentSizeBytes();
+
+  /**
+   * Get the storage tier of the immutable segment.
+   *
+   * @return storage tier, null by default.
+   */
+  default String getTier() {
```

Comment:
Annotate it as `nullable`, and suggest not adding a default implementation here because we want all implementations to explicitly implement this